parachain/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -418,17 +418,18 @@ pub mod pallet {
 			// Though following check does not belong to ALC spec we verify block_roots_root to
 			// match the finalized checkpoint root saved in the state of `finalized_header` so to
 			// cache it for later use in `verify_ancestry_proof`.
-			ensure!(
-				verify_merkle_branch(
-					update.block_roots_root,
-					&update.block_roots_branch,
-					config::BLOCK_ROOTS_SUBTREE_INDEX,
-					config::BLOCK_ROOTS_DEPTH,
-					update.finalized_header.state_root
-				),
-				Error::<T>::InvalidBlockRootsRootMerkleProof
-			);
-
+			if !update.block_roots_branch.is_empty() {
+				ensure!(
+					verify_merkle_branch(
+						update.block_roots_root,
+						&update.block_roots_branch,
+						config::BLOCK_ROOTS_SUBTREE_INDEX,
+						config::BLOCK_ROOTS_DEPTH,
+						update.finalized_header.state_root
+					),
+					Error::<T>::InvalidBlockRootsRootMerkleProof
+				);
+			}
```

Comment:
For the reason mentioned in https://github.com/Snowfork/snowbridge/pull/1097#discussion_r1441432101

Reply:
Hmmm but how can this merkle proof be conditional? Could it not lead to unexpected behaviour? As an example, this check allows finalized header verification without providing the block roots proof. What could happen is we keep processing these beacon headers without the necessary ancestry proofs (needed for the execution header imports). Then, our execution header imports could fail because there are no ancestry proofs to verify the execution header against a beacon header.

Reply:
https://github.com/Snowfork/snowbridge/pull/1097#discussion_r1444244154